podpac/core/interpolation/interpolation.py

```diff
@@ -26,12 +26,11 @@ def interpolation_decorator():
 class InterpolationMixin(tl.HasTraits):
     interpolation = InterpolationTrait().tag(attr=True)
 
-    @node_eval
-    def eval(self, coordinates, output=None, selector=None):
+    def _eval(self, coordinates, output=None, _selector=None):
         node = Interpolate(interpolation=self.interpolation)
         node._set_interpolation()
-        node._source_xr = super().eval(coordinates, selector=node._interpolation.select_coordinates)
-        return node.eval(coordinates, output)
+        node._source_xr = super()._eval(coordinates, _selector=node._interpolation.select_coordinates)
```

Comment:
@mpu-creare is this really working? Doesn't `super()` (without arguments) not work in python 2? Is this something we missed in the last merge?

Reply:
super() without arguments do NOT work in Python 2. We've dropped Python 2 support. :)

Reply:
Oh wow, somehow I missed that. Okay! There's a bit of cruft here and there that can be cleaned up, in that case. Should I do that at some point?

Reply:
I actually prefer the super without the class name... that way if you ever change the class name you don't get STRANGE errors.

Reply:
I prefer the super without the class name, too. I would update all of the super calls to be that way. In addition:

```
podpac/core/managers/test/test_multithreading.py:        if sys.version_info.major == 2:
podpac/core/managers/test/test_multithreading.py:            if sys.version_info.major == 2:
podpac/core/managers/multi_process.py:        if (sys.version_info.major + sys.version_info.minor / 10.0) >= 3.7:
podpac/core/utils.py:if sys.version < "3.6":
podpac/core/data/file_source.py:if sys.version_info.major == 2:
podpac/core/algorithm/test/test_generic.py:if sys.version_info.major == 2:
podpac/core/algorithm/generic.py:if sys.version_info.major == 2:
podpac/core/test/test_utils.py:    @pytest.mark.skipif(sys.version < "3.6", reason="python < 3.6")
podpac/core/test/test_utils.py:    @pytest.mark.skipif(sys.version >= "3.6", reason="python >= 3.6")
podpac/core/__init__.py:if sys.version_info.major == 2:
podpac/core/coordinates/base_coordinates.py:    if sys.version < "3":
podpac/core/coordinates/coordinates.py:        if sys.version < "3.6":
podpac/core/coordinates/coordinates.py:    if sys.version < "3":
podpac/core/coordinates/test/test_coordinates.py:        if sys.version < "3.6":
podpac/datalib/weathercitizen_sensorburst_pb2.py:_b = sys.version_info[0] < 3 and (lambda x: x) or (lambda x: x.encode("latin1"))
podpac/datalib/__init__.py:if sys.version >= "3.6":
podpac/__init__.py:if sys.version_info.major == 2:
setup.py:if sys.version_info.major == 2:
setup.py:if sys.version_info.major == 2:
setup.py:if sys.version >= "3.6":
```

Reply:
I'm not ready to clean that up just yet... I'm waiting to see how things shake out with another project in case we need to go back to Python 2 for some reason... I really don't think so, but once we get past that hurdle we can clean that up.